src/odemis/gui/cont/tabs.py

```diff
@@ -3918,78 +3922,188 @@ def _onClickFocus(self, evt):
+            # Fixed values, known to work well for autofocus
+            speclines.detExposureTime.value = speclines.detExposureTime.clip(0.2)
+            self._setFullFoV(speclines, (2, 2))
+            if model.hasVA(speclines, "detReadoutRate"):
+                try:
+                    speclines.detReadoutRate.value = speclines.detReadoutRate.range[1]
+                except AttributeError:
+                    speclines.detReadoutRate.value = max(speclines.detReadoutRate.choices)
+            focus_streams.append(speclines)
+
+        return focus_streams
+
+    def _ensureOneFocusStream(self, _):
+        """
+        Ensures that only one focus stream is shown at a time
+        Called when a focus stream "should_update" state changes
+        Add/Remove the stream to the current view (updating the StreamTree)
+        Set the focus detectors combobox selection to the the stream's detector
+        # Pick the stream to be shown:
+        #  1. Pick the stream which is playing (should_update=True)
+        #  2. Pick the focus stream which is already shown (in v.getStreams())
+        #  3. Pick the first focus stream
+        """
+        focusedview = self.tab_data_model.focussedView.value
+        should_update_stream = next((s for s in self._focus_streams if s.should_update.value == True), None)
+        if should_update_stream:
+            # This stream should be shown, remove the other ones first
+            for st in focusedview.getStreams():
+                if st in self._focus_streams and st is not should_update_stream:
+                    focusedview.removeStream(st)
+            focusedview.addStream(should_update_stream)
+            # Set the focus detectors combobox selection
+            try:
+                istream = self._focus_streams.index(should_update_stream)
+                if istream != -1:
+                    self.panel.cmb_focus_detectors.SetSelection(istream)
+            except ValueError:
+                logging.error("Unable to find index of the focus stream")
+        else:
+            # Check if there are any focus stream currently shown
+            if not any(s in self._focus_streams for s in focusedview.getStreams()):
+                # Otherwise show the first one
+                focusedview.addStream(self._focus_streams[0])
+                self.panel.cmb_focus_detectors.SetSelection(0)
+
+    def _onFocusDetectorChange(self, evt):
+        """
+        Handler for focus detector combobox selection change
+        Play the stream associated with the chosen detector
+        """
+        # Find the stream related to this selected detector
+        idx = self.panel.cmb_focus_detectors.GetSelection()
+        stream = self._focus_streams[idx]
+        # Play this stream (set both should_update and is_active with True)
+        stream.should_update.value = True
+        stream.is_active.value = True
+
+    def add_combobox_control(self, label_text, value=None, conf=None):
```

Comment:
Maybe this method should be private?

Reply:
It needs to be the same signature to be used correctly by `create_axis_entry`.